cli/chaosli/cmd/explain.go

```diff
@@ -7,26 +7,245 @@ package cmd
+		fmt.Printf("\t\t\t⛵️ Port: %d\n", data.Port)
+		fmt.Printf("\t\t\t🧾 Protocol: %s\n", data.Protocol)
+	}
+
+	fmt.Println("\t💥  will apply filters so that network failures apply to outgoing/ingoing traffic from/to the following services/namespaces pairs:")
+
+	for _, data := range network.Services {
+		fmt.Printf("\t\t🎯 Service: %s\n", data.Name)
+		fmt.Printf("\t\t\t⛵️ Namespace: %s\n", data.Namespace)
+	}
+
+	if network.Flow == v1beta1.FlowIngress {
+		fmt.Println("\t💥 applies network failures on incoming traffic instead of outgoing.")
+	} else {
+		fmt.Println("\t💥 applies network failures on outgoing traffic.")
+	}
+
+	if network.Drop != 0 {
+		fmt.Printf("\t\t💣 applies a packet drop of %d percent.\n", network.Drop)
+	}
+
+	if network.Corrupt != 0 {
+		fmt.Printf("\t\t💣 will corrupt packets at %d percent.\n", network.Drop)
+	}
+
+	if network.Delay != 0 {
+		fmt.Printf("\t\t💣 applies a packet delay of %d ms.\n", network.Delay)
+
+		if network.DelayJitter != 0 {
+			fmt.Printf("\t\t\t💣 applies a jitter of %d ms to the delay value to add randomness to the delay.\n", network.Delay)
+		}
+	}
+
+	if network.BandwidthLimit != 0 {
+		fmt.Printf("\t\t💣 applies a bandwidth limit of %d ms.\n", network.BandwidthLimit)
+	}
+
+	printSeparator()
+}
+
+func explainMultiDisruption(spec v1beta1.DisruptionSpec) {
+	existsMulti := false
+
+	if spec.NodeFailure != nil {
+		if spec.CPUPressure != nil || spec.DNS != nil || spec.DiskPressure != nil || spec.Network != nil {
+			fmt.Println("⚠️  You are attempting to run a Node Failure Disruption in addition to another one of our other failures.\n" +
+				"   Keep in mind that once the Node Failure runs (the kernel panic) the other disruptions will most likely not.")
+
+			existsMulti = true
+		}
+	}
+
+	if spec.Network != nil && spec.CPUPressure != nil {
+		fmt.Println("⚠️  You are attempting to run a Network Disruption and a CPU Pressure Disruption.\n" +
+			"   Keep in mind that CPU Pressure will most likely add additional issues to the network therefore your network disruption\n" +
+			"   will be less exact in its defined failures.")
+
+		existsMulti = true
+	}
+
+	if existsMulti {
+		printSeparator()
+	}
+}
+
 var explainCmd = &cobra.Command{
 	Use:   "explain",
 	Short: "explains disruption config",
 	Long:  `translates the yaml of the disruption configuration to english.`,
 	Run: func(cmd *cobra.Command, args []string) {
 		path, _ := cmd.Flags().GetString("path")
-		fmt.Println(explanation(path))
+		explanation(path)
 	},
 }
 
-func explanation(filePath string) string {
+func explanation(filePath string) {
+	var disruption v1beta1.Disruption
+
 	if filePath == "" {
-		return pathError
+		fmt.Println(pathError)
+	}
+
+	fullPath, _ := filepath.Abs(filePath)
+	disruptionBytes, err := ioutil.ReadFile(fullPath)
+
+	if err != nil {
+		log.Printf("disruption.Get err   #%v ", err)
+		os.Exit(1)
 	}
 
-	return "Explanation TODO"
+	err = yaml.Unmarshal(disruptionBytes, &disruption)
```

Comment:
It would be good to print out the validation output too whenever we call explain, wdyt?

Reply:
Yes, debating if we should even run explain on invalid disruptions. Would save us a lot of headache in the explain methodology all together.